compiler-rt/test/tsan/ignore_lib0.cpp

```diff
@@ -4,11 +4,12 @@
 // RUN: %clangxx_tsan -O1 -fno-builtin %s -DLIB -fPIC -fno-sanitize=thread -shared -o %t-dir/libignore_lib0.so
 // RUN: %clangxx_tsan -O1 %s -L%t-dir -lignore_lib0 %link_libcxx_tsan -o %t
 // RUN: echo running w/o suppressions:
-// RUN: env LD_LIBRARY_PATH=%t-dir${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH} %deflake %run %t | FileCheck %s --check-prefix=CHECK-NOSUPP
+// RUN: bash -c env | grep LD_LIBRARY_PATH | sed s/.*LD_LIBRARY_PATH=// > %t2
```

Comment:
don't we have a way to read the environment in the built in shell? This doesn't seem to reproduce what the original code did (not use `:` when `LD_LIBRARY_PATH` is unset)

Reply:
The implementation of 'env' in the internal shell does not allow you to pipe it to other commands such as 'grep'. I was working on an implementation to fix that, but have not been able to get it to work yet.

When working on this test, I threw in some extra 'env' calls (temporarily) and looked at the output, both with the external shell and the internal shell, and made sure the LD_LIBRARY_PATH definitions we got in the internal shell were the same as the ones we got in the external shell.

Reply:
Hm...I think the environments in which I ran my tests always had LD_LIBRARY_PATH defined by default.  Let me go back and work on this a bit more (thank you for pointing this out).

Reply:
Just wondering (since fixing this looks difficult): Does having a trailing colon, in the case where LD_LIBRARY_PATH was not previously set, actually cause anything to break?

Reply:
It causes a difference in that it adds the current directory to the search path:

```
$ touch libc.so.6
$ LD_LIBRARY_PATH=/tmp /bin/true
$ LD_LIBRARY_PATH=/tmp: /bin/true
/bin/true: error while loading shared libraries: libc.so.6: file too short
```

Reply:
Maybe we can add `/dev/null` instead if it's empty?

Reply:
Or wouldn't `sed s/.*LD_LIBRARY_PATH=/:/` give us the old behavior?

Reply:
It ought to, but I'm still having problems with the internal shell. I'm going to revert this test for now, so the rest of the PR can be committed, while I continue debugging this in the internal shell.